cmd/nvidia-device-plugin/plugin-manager.go

```diff
@@ -39,15 +39,20 @@ func NewPluginManager(infolib info.Interface, nvmllib nvml.Interface, devicelib
 		return nil, fmt.Errorf("unknown strategy: %v", *config.Flags.MigStrategy)
 	}
 
+	// TODO: We could consider passing this as an argument since it should already be used to construct nvmllib.
+	driverRoot := root(*config.Flags.Plugin.ContainerDriverRoot)
+
 	deviceListStrategies, err := spec.NewDeviceListStrategies(*config.Flags.Plugin.DeviceListStrategy)
 	if err != nil {
 		return nil, fmt.Errorf("invalid device list strategy: %v", err)
 	}
 
 	cdiHandler, err := cdi.New(infolib, nvmllib, devicelib,
 		cdi.WithDeviceListStrategies(deviceListStrategies),
-		cdi.WithDriverRoot(*config.Flags.Plugin.ContainerDriverRoot),
+		cdi.WithDriverRoot(string(driverRoot)),
+		cdi.WithDevRoot(driverRoot.getDevRoot()),
 		cdi.WithTargetDriverRoot(*config.Flags.NvidiaDriverRoot),
+		cdi.WithTargetDevRoot(*config.Flags.NvidiaDevRoot),
```

Comment:
Note that the `TargetDevRoot` is the dev root on the host. This is set to `NvidiaDevRoot`.